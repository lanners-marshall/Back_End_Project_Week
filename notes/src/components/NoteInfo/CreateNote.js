import React from 'react';
import axios from 'axios';
import { BGColor, SubmitContainer, InputTitle, InputContent, DivClick, Flexdiv, Red, SelectDiv} from './css';
import Select from 'react-select';

console.log()

class CreateNote extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			title: '',
			text: '',
			author: '',
			errorTitle: '',
			errorBody: '',
			collaborators: '',
			selectedOption: [],
		};
	}

	componentDidMount(){
		const token = localStorage.getItem('jwt')
		const reqOptions = {
			headers: {
				Authorization: token,
			}
		};

		axios
			.get('http://localhost:5555/notes', reqOptions)
			.then(response => {
				//console.log(response)
			})
			.catch(error => {
				//console.log(error)
				localStorage.setItem("error", "Please Log In");
				this.props.history.push('/signin')
			})

		axios
			.get('http://localhost:5555/collaborators')
			.then(response => {
				this.setState({
					collaborators: response.data
				})
			})
			.catch(error => {
				//console.log(error)
			})
	}

  handleSelect = (selectedOption) => {
    this.setState({ selectedOption });
    //console.log(`Option selected:`, selectedOption);
  }

	createNote = () => {

		let obj = {};
		let collabs = this.state.selectedOption
		
		for (let i = 0; i < this.state.collaborators.length; i++){
			if (this.state.collaborators[i].name === localStorage.getItem('loggedInAs')){
				obj = {value: this.state.collaborators[i].id, label: localStorage.getItem('loggedInAs') }
			}
		}

		collabs.push(obj);

		const note = {
			title: this.state.title, 
			text: this.state.text, 
			author: localStorage.getItem('loggedInAs'),
			collaborators: collabs,
		};

		axios
			.post("http://localhost:5555/notes", note)
			.then(response => {
				console.log("post response: ", response);
				this.setState({title: '', text: ''})

				//I will update state to show the new post in view
				axios
					.get("http://localhost:5555/notes")
					.then(response => {
						this.props.handleData(response.data)
					})
					.catch(err => {
						console.log(err)
					})
			})
			.catch(error => console.log(error));
			this.props.history.push("/notes")

// [
// 	{
// 		label: “sam”
// 		value: 2
// 	}
// 	{
// 		label: “carl”
// 		value: 2
// 	}
// 	{
// 		label: “Timmy”
// 		value: 5
// 	}
// ]

		//{note_id: 1, collaborator_id: 2}, // jessica

		//need a post to notes_collaborators table
		// axios
		// 	.post
	}

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value})
  }

	render(){		

		//console.log(this.props)
		let currentUser = localStorage.getItem('loggedInAs')
		let options = []
		//console.log(this.state.collaborators)
		for (let i = 0; i < this.state.collaborators.length; i++){
			//filter out who is currently logged in from list of users
			if (this.state.collaborators[i].name === currentUser){
				{continue}
			} else {
				options.push({value: this.state.collaborators[i].id, label: this.state.collaborators[i].name})
			}
		}

		// console.log(options)

		const { selectedOption } = this.state;
		return (
			<BGColor>
				<SubmitContainer>
					<h3>Create New Note:</h3>
					<form>
						<InputTitle
							type="text"
							placeholder='Note Title'
							onChange={this.handleChange}
							name="title"
							value={this.state.title}
							maxLength="30"
						/>
						<InputContent
							type="text"
							placeholder='Note Content'
							onChange={this.handleChange}
							name="text"
							value={this.state.text}
							maxLength="229"
						/>
					</form>
					<SelectDiv>
						<p>Select Collaborators</p>
						<Select
			        value={selectedOption}
			        onChange={this.handleSelect}
			        options={options}
			        isMulti={true}
			        isSearchable={true}
			      />
		      </SelectDiv>

					<Flexdiv>
						<DivClick onClick={this.createNote}>Save</DivClick>
					</Flexdiv>
				</SubmitContainer>
			</BGColor>
		)
	}
}

export default CreateNote;
