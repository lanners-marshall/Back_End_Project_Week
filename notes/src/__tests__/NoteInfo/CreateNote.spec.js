import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CreateNote from '../../components/NoteInfo/CreateNote';
import renderer from 'react-test-renderer';

describe('<CreateNote />', () => {
  it('should match snapshot', () => {
	  const wrapper = shallow(
	      <CreateNote />
  	)
  	const tree = renderer.create(wrapper).toJSON()
  	expect(tree).toMatchSnapshot()
	})
});
