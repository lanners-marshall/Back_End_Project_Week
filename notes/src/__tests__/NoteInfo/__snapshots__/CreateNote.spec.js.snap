// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateNote /> should match snapshot 1`] = `
<div
  className="sc-bdVaJa mMasL"
>
  <div
    className="sc-bwzfXH jJIZoS"
  >
    <h3>
      Create New Note:
    </h3>
    <form>
      <input
        className="sc-htpNat jncqSh"
        maxLength="30"
        name="title"
        onChange={[Function]}
        placeholder="Note Title"
        type="text"
        value=""
      />
      <textarea
        className="sc-bxivhb euUZsj"
        maxLength="229"
        name="text"
        onChange={[Function]}
        placeholder="Note Content"
        type="text"
        value=""
      />
    </form>
    <div
      className="sc-EHOje dISPpS"
    >
      <button
        className="sc-ifAKCX dCxPJL"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
